### PR TITLE
Auto-encode audio track for Whisper

### PR DESCRIPTION
**Pull request by lkiesow**
Whisper.cpp requires a very specific input audio format. Instead of relying on adopters working around this in the workflow and encoding tracks in advance, this patch includes the encoding in the operation itself.

This not only has the advantage that the configuration is much easier, but it also means that only the tracks selected to actually be used for generating the subtitles are encoded.

### How to test this patch

Configure Whisper.ccp and add the `speechtotext` operation to the fast testing workflow:

```yaml
  - id: speechtotext                                                                                                     
    description: Generates subtitles for video and audio files                                                           
    configurations:                                                                                                      
      - source-flavor: '*/source'                                                                                        
      - target-flavor: captions/source                                                                                   
      - target-tags: engage-download                                                                                     
      - target-flavor: captions/source                                                                                   
      - limit-to-one: true 
```

If you now ingest an `.mp4` file, the operation should be able to work on it.

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://docs.opencast.org/develop/developer/#participate/development-process/#automatically-closing-issues-when-a-pr-is-merged) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)
* [x] explain why it needs to be merged into the legacy branch, if it is targeting the legacy branch

### CODE REVIEW

File: etc/org.opencastproject.speechtotext.impl.engine.WhisperCppEngine.cfg [hunk 1]
```diff
@@ -1,12 +1,18 @@
+<<<<<<< HEAD
 # Configuration for setting a custom path to the Whisper command line tool
 # Default: whisper
 #whispercpp.root.path=whispercpp
+=======
+# Configuration for setting a custom path to the Whisper.cpp command line tool
+# Default: whisper.cpp
+whispercpp.root.path=/home/lars/dev/whisper.cpp/main
+>>>>>>> f7b28fafd2 (Auto-encode audio track for Whisper)
```
Comment (geichelberger) [inline etc/org.opencastproject.speechtotext.impl.engine.WhisperCppEngine.cfg]: Unresolved merge conflict residues.

**Review by geichelberger — DISMISSED** [review]

File: etc/org.opencastproject.speechtotext.impl.engine.WhisperCppEngine.cfg (hunk 1 above)
Comment (lkiesow) [reply etc/org.opencastproject.speechtotext.impl.engine.WhisperCppEngine.cfg]: Ups, that's a leftover from some local tests.
Fixed.

**Review by marwyg — APPROVED** [review]
I tested following scenarios with the fast Workflow:
- auto encode: true -> works as expected
- auto encode: false + .mp4 file -> fails as expected saying "wrong media extension"
- auto encode: false + .wav file -> works as expected

The last 2 test did ensure, that the behavior stays like before, when setting the config to false. But the default is true, so I guess the target branch (develop) is okay then? I like the idea, that auto-encode should be the default behavior. So for my part it can stay like this.

I guess it would be nice if this feature would be mentioned somewhere in the release notes? If adopter use whispercpp and doesn't change this, it would not break something I think. There would be only an unnecessary encoding operation in front of their speechtotext operation.

Regarding the code, I have nothing to complain about.

**Comment by mtneug** [comment]
> I guess it would be nice if this feature would be mentioned somewhere in the release notes?

+0

**Comment by lkiesow** [comment]
> I guess it would be nice if this feature would be mentioned somewhere in the release notes? 

I added a note to `docs/guides/admin/docs/releasenotes`.
I hope this can be merged like this.

**Comment by lkiesow** [comment]
@marwyg, are you okay with the current state and can this be merged?

**Comment by marwyg** [comment]
Yes, everything is fine. I don't see anything that would speak against a merge.

**Comment by KatrinIhler** [comment]
In that case...lgtm

**Pull request merged**